### PR TITLE
tests: Rewrite updatePKUK to avoid outputing a lot of messages about generating PKs

### PR DESCRIPTION
**Pull request by suzaku**
### What problem does this PR solve? 

updatePKUK output a lot of logs for retrying to generate unused PKs.

### What is changed and how it works?

Record free PKs explicitly to avoid retrying.

### Check List 

Tests 

 - Integration test

Code changes

Side effects

Related changes

### CODE REVIEW

File: tests/run.sh [hunk 1]
```diff
@@ -161,7 +161,7 @@ run_case() {
 }
 
 # List the case names to run, eg. ("binlog" "kafka")
-do_cases=()
+do_cases=("binlog")
```
Comment (july2993) [inline tests/run.sh]: --?

Comment (suzaku) [reply tests/run.sh]: Fixed

File: tests/dailytest/case.go [hunk 2]
```diff
@@ -351,34 +351,40 @@ func updatePKUK(db *sql.DB, opNum int) error {
 	maxKey := 20
 	mustExec(db, "create table pkuk(pk int primary key, uk int, v int, unique key uk(uk));")
 
-	var pks []int
+	pks := make(map[int]struct{})
+	freePks := make([]int, maxKey)
```
Comment (lichunzhu) [inline tests/dailytest/case.go]: How about `freePks := rand.Perm(maxKey)`?

Comment (suzaku) [reply tests/dailytest/case.go]: Cool, I didn't know about that, thanks.

**Comment by lichunzhu** [comment]
LGTM

File: tests/dailytest/case.go [hunk 3]
```diff
@@ -351,34 +351,37 @@ func updatePKUK(db *sql.DB, opNum int) error {
 	maxKey := 20
 	mustExec(db, "create table pkuk(pk int primary key, uk int, v int, unique key uk(uk));")
 
-	var pks []int
+	pks := make(map[int]struct{})
+	freePks := rand.Perm(maxKey)
+
+	nextPk := func() int {
+		return freePks[0]
```
Comment (july2993) [inline tests/dailytest/case.go]: https://godoc.org/math/rand#Shuffle before return first one or randomly pick one?

Comment (suzaku) [reply tests/dailytest/case.go]: @july2993 `rand.Perm` is enough? On deletion, the pks are deleted in random order and appended back to this slice.

Comment (july2993) [reply tests/dailytest/case.go]: there's no way you will do thinks like this:
update pkuk set pk = 1 where pk = 2;
update pkuk set pk = 2 where pk = 1;
update pkuk set pk = 1 where pk = 2;
update pkuk set pk = 2 where pk = 1;
...
continually to swap the two values, why not just simply make it more randomly?

**Pull request merged**